### DISCUSSION TITLE
Override commit message suffix

### DISCUSSION THREAD

**Discussion by henrygriffiths** [renovatebot/renovate#16886] [Request Help]
Hi All,

I maintain a repository that occasionally has long lived feature branches that need the use of renovate to keep dependencies up to date. So far, setting `baseBranches` to contain both `develop` (default branch) and `feature-branch` works like a charm. However, we don't always have a long lived feature branch that is in progress. This results in difference in PR and subsequently commit titles, with some containing a suffix of `(develop)` and some not, and feature branches always containing a suffix of `(feature-branch)`. Is it possible to configure renovate such that the `commitMessageSuffix` for the `develop` branch is always empty and the commit message suffix for feature branches is always the feature branch?

So far I've tried (abbreviated):
```json
{
  "extends": [
    "config:base",
    ":semanticCommitTypeAll(chore)"
  ],
  "baseBranches": [
    "develop",
    "feature-branch"
  ],
  "packageRules": [
    {
      "matchBaseBranches": ["develop"],
      "commitMessageSuffix": ""
    },
    {
      "matchBaseBranches": ["feature-branch"],
      "commitMessageSuffix": "(feature-branch)"
    }
  ]
}
```

However, this results in PRs for the `develop` branch to still have the `(develop)` suffix, and the suffix for the `feature-branch` branch to have `(feature-branch) (feature-branch)`.
I've also looked at editing the `commitMessage` option, however, it is noted that this is deprecated, and not recommended, and I'm not sure how to also configure it to support vulnerability alerts on different branches (using the `useBaseBranchConfig` option seems like a possibility if all else fails, but would be a bit of a management nightmare given the use case here).

Any thoughts?

**Comment by PhilipAbed** [comment]
[commitMessage](https://docs.renovatebot.com/configuration-options/#commitmessage) is made of :
`{{{commitMessagePrefix}}} {{{commitMessageAction}}} {{{commitMessageTopic}}} {{{commitMessageExtra}}} {{{commitMessageSuffix}}}`

so one of the parts is giving you `(develop)` and `(feature-branch)`, find which one of the above variables it is
according to your tests it's not `commitMessageSuffix`
it could be the `commitMessageTopic`:

try this 
```
 "packageRules": [
    {
      "matchBaseBranches": ["develop"],
      "commitMessageTopic": ""
    }
  ]
```

**Reply by henrygriffiths** [reply]
Thanks for the quick reply @PhilipAbed.

Per your suggestion of setting `commitMessageTopic` to `""` results in commits like `chore: update to v1.2.3 (develop)`.

Despite the warnings, editing the `commitMessage` to `{{{commitMessagePrefix}}} {{{commitMessageAction}}} {{{commitMessageTopic}}} {{{commitMessageExtra}}}` results in the duplicate `(feature-branch) (feature-branch)` being dropped, but still results in the original behavior (aka without setting `commitMessage`) at all.

**Reply by PhilipAbed** [reply]
That means the `commitMessageSuffix` is being automatically defaulted to `(feature-branch)`  because you are using multiple branches

**Reply by henrygriffiths** [reply]
Looking at it further and doing a few more tests, I believe this is outside of the `commitMessage` (and any sub values), and within the `prTitle`. When changing these values, I see the values in the dependency dashboard get changed, and these reflect the PR Titles when they get created, however, the commit message in the PR does not contain the branch despite the title containing it (and any `commitMessageSuffix` value, such as `(feature-branch)` that is added in the config is reflected in both the title and commit message. 

As such, I've set `prTitle` to null, however this does not result in a change to the PR Titles as seen in the dependency dashboard.

**Reply by PhilipAbed** [reply]
nothing about `commitMessageSuffix` in docs: https://docs.renovatebot.com/configuration-templates/#commit-message that does that
cant see anything that is hardcoding it in code either 

according to docs : 

[PR Title](https://docs.renovatebot.com/configuration-templates/#pr-title)
```
Because commit messages match with the PR title, the PR title template defaults to null and inherits/copies the value from 
commitMessage. If you have a requirement where prTitle should be different from commitMessage, then please [raise a feature request]
(https://github.com/renovatebot/renovate/issues) for discussion.
```

**Reply by viceice** [reply]
https://github.com/renovatebot/renovate/blob/0f0f2f79ac6ff894933f8304c9bd9c0bbe52fd1a/lib/workers/repository/updates/generate.ts#L240

**Reply by viceice** [reply]
https://github.com/renovatebot/renovate/blob/0f0f2f79ac6ff894933f8304c9bd9c0bbe52fd1a/lib/workers/repository/process/index.ts#L64-L67

**Reply by henrygriffiths** [reply]
From my tests, the commit message is doing exactly what I want it to do (albeit having to manually specify the `commitMessageSuffix` for each feature branch, it's the title that is different (containing the branch name), which is the issue, as that's actually what gets used when the PR is merged in my scenario.

According to https://docs.renovatebot.com/configuration-options/#prtitle, setting `prTitle` to null _should_ be the solution, as I can now easily set the `commitMessage` to exactly what I want, however, that doesn't seem to work and results in a warning in the dependency dashboard.

**Reply by henrygriffiths** [reply]
I did appear to miss the `deprecated` notice on the `prTitle` config setting, which explains why it isn't working. From the code snippets provided, I cannot seem to find a way to override said behavior of appending the `baseBranch` to `upgrade.prTitle` when multiple base branches are provided, but I'm not sure if I'm missing anything there.
